sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py

```diff
@@ -141,9 +161,12 @@ def __init__(
         self.primary_user_assigned_identity = primary_user_assigned_identity
         self.managed_network = managed_network
         self.enable_data_isolation = enable_data_isolation
-        self.workspace_hub = workspace_hub
-        if workspace_hub:
-            self._kind = PROJECT_WORKSPACE_KIND
+        self.__hub_id = hub_id
```

Comment:
Do you have double underscore here?

Reply:
Yes, due to mypy check issues. TL;DR mypy demands that the base workspace class surface `_hub_id` in order for some polymorphism to work properly in the operations logic. That means that the 'real' hub id needs to be double underscored in order to be distinct from the surfaced, single-underscore variant.

Reply:
Somewhat related to my thread, is there a reason you couldn't just make `__hud_id` into `_hub_id` and get rid of the property? Does mypy explicitly want a property and not an attribute?

Reply:
Yes, Mypy explicitly wanted a property. I initially tested just setting self._hub_id and not publisizing with a property, but mypy was not having it.